forge/src/multi_runner.rs

```diff
@@ -195,14 +185,14 @@ impl MultiContractRunner {
         filter: &(impl TestFilter + Send + Sync),
         stream_result: Option<Sender<(String, BTreeMap<String, TestResult>)>>,
     ) -> Result<BTreeMap<String, BTreeMap<String, TestResult>>> {
-        let source_paths = self.source_paths.clone();
         let env = self.evm_opts.evm_env();
         let results = self
             .contracts
             .par_iter()
-            .filter(|(name, _)| filter.matches_path(source_paths.get(*name).unwrap()))
-            .filter(|(name, _)| filter.matches_contract(name))
-            .filter(|(_, (abi, _, _))| abi.functions().any(|func| filter.matches_test(&func.name)))
```

Comment:
This check was mistakenly duplicated here by me

Reply:
Ah, I remember now - I do another similar check in `ContractRunner`, this one is to avoid deploying contracts we aren't going to test anyway (the check in `ContractRunner` is slightly different on master, so the check here is not on master)